packages/engine/src/createLink.js

```diff
@@ -16,32 +16,42 @@
 
 import { type, urlQuery as urlQueryFn } from '@lowdefy/helpers';
 
-function createLink({ backLink, lowdefy, newOriginLink, sameOriginLink }) {
-  function link({ back, home, input, newTab, pageId, url, urlQuery }) {
-    let pathname = pageId;
-    if (back) {
-      return backLink();
+function createLink({ backLink, disabledLink, lowdefy, newOriginLink, noLink, sameOriginLink }) {
+  function link(props) {
+    if (props.disabled === true) {
+      return disabledLink(props);
     }
-    const lowdefyUrlQuery = type.isNone(urlQuery) ? '' : `?${urlQueryFn.stringify(urlQuery)}`;
-    if (home) {
-      if (lowdefy.home.configured) {
-        pathname = '';
-        pageId = lowdefy.home.pageId;
-      } else {
-        pathname = lowdefy.home.pageId;
-        pageId = lowdefy.home.pageId;
-      }
+    if ([!props.pageId, !props.back, !props.home, !props.url].filter((v) => !v).length > 1) {
+      throw Error(
+        `Invalid Link: To avoid ambiguity, only one of 'back', 'home', 'pageId' or 'url' can be defined.`
+      );
     }
-    if (!type.isNone(pathname)) {
-      if (!type.isNone(input)) {
-        lowdefy.inputs[pageId] = input;
-      }
-      return sameOriginLink(`/${pathname}${lowdefyUrlQuery}`, newTab);
+    if (props.back === true) {
+      // Cannot set input or urlQuery on back
+      return backLink(props);
     }
-    if (!type.isNone(url)) {
-      return newOriginLink(`${url}${lowdefyUrlQuery}`, newTab);
+    const lowdefyUrlQuery = type.isNone(props.urlQuery)
+      ? ''
+      : `?${urlQueryFn.stringify(props.urlQuery)}`;
+    if (props.home === true) {
+      lowdefy.inputs[`page:${lowdefy.home.pageId}`] = props.input;
```

Comment:
So this will set input to undefined or null if it isn't defined here, but it was set before. I think that is the expected behaviour, but this is a breaking change from:
```js
      if (!type.isNone(input)) {
         lowdefy.inputs[pageId] = input;
       }
```
Maybe we should also set a empty object default here, or ensure the null is correctly handled in engine because I don't think engine checks this except when the context is created

Reply:
I wouldn't set it here, but rather default during read, which is think is handled by `get`. When looking at `lowdefy.inputs` it's nice to see only inputs that were set explicitly. Else we might default to an empty object and make it hard to inspect why the value was not passed down. In short, undefined is better for debugging in my opinion. Thoughts?